library/init/meta/tactic.lean

```diff
@@ -1004,8 +1004,10 @@ do tgt : expr ← target,
 meta def by_cases (e : expr) (h : name) : tactic unit :=
 do dec_e ← (mk_app `decidable [e] <|> fail "by_cases tactic failed, type is not a proposition"),
    inst  ← (mk_instance dec_e <|> fail "by_cases tactic failed, type of given expression is not decidable"),
-   em    ← mk_app `decidable.em [e, inst],
-   cases em [h, h]
+   t     ← target,
+   tm    ← (mk_mapp `dite [some e, some inst, some t] <|>
+     fail "by_cases tactic failed, type of given expression is not decidable"),
```

Comment:
I don't think we can ever hit this case.  If the proposition is not decidable, already the call to mk_instance will have failed.